### PR TITLE
Base TRE API

### PR DESCRIPTION
**Pull request by TessFerrandez**
# Build base structure for TRE API

## What is being addressed

- Add TRE API base structure
- Implement ping endpoint - returns pong
- Add python linting

## How is this addressed

- API implemented with FastAPI under core/api (idea is to have all python applications under core)
- Endpoints implemented under core/api/routers folder (models will eventually go under models)
- API tests with test for ping endpoint in core/api_tests - executable with pytest
- README.md with information on how to run and test locally
- Add flake8 linting to develop and main branch on PRs
- Add optional pre-commit hook for flake8 linting
- Add documentation for pre-commit

### CODE REVIEW

File: core/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+virtualenv venv
+source venv/bin/activate    # linux
+venv/Scripts/activate       # windows
+pip install -r core/api/requirements.txt
+```
+
+Run the webserver locally - from a terminal in the core directory
+
+```cmd
+uvicorn api.main:app
```
Comment (marrobi) [inline core/README.md:25]: Caught me out that hadn't changed directory (RTFM). Could a `cd core` or something in the command help?

**Review by marrobi — APPROVED** [review]
One small comment, tested and works, nice to see some code written!

File: core/README.md (hunk 1 above)
Comment (TessFerrandez) [reply core/README.md:25]: will add that :)

**Comment by deniscep** [comment]
Tested on app service, works, looks good.

**Review by deniscep — APPROVED** [review]

**Pull request merged**